helpers/url.ts

```diff
@@ -2,6 +2,17 @@ import {ensureNonEmptyString} from './ensure';
 import qs from 'qs';
 import urlParse from 'url-parse';
 
+/**
+ * Helper to create a new URL by appending parameters to a base URL.
+ *
+ * The input URL may or may not having existing parameters.
+ *
+ * @example
+ * ```
+ * // Returns `"/someApi/someEndpoint?token=asdf&limit=5"`
+ * const url = withQueryParams("/someApi/someEndpoint", {token: "asdf", limit: 5});
```

Comment:
Nitpick: The style guide says to prefer `let` over `const` except when it's a file-level constant (which this URL likely isn't).

Reply:
Fixed.